From 9689286b990709414227030a13ac2f287fed5a9c Mon Sep 17 00:00:00 2001
From: Sven Neumann <s.neumann@raumfeld.com>
Date: Mon, 9 Jul 2012 12:13:48 +0200
Subject: [PATCH] xdgmime: return early for empty data

Add a check for the data length to xdg_mime_get_mime_type_for_data()
and return early for len == 0 instead of crashing later in
cache_magic_matchlet_compare().

Signed-off-by: Sven Neumann <s.neumann@raumfeld.com>
---
 gio/xdgmime/xdgmime.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/gio/xdgmime/xdgmime.c b/gio/xdgmime/xdgmime.c
index 846be39..9e53500 100644
--- a/gio/xdgmime/xdgmime.c
+++ b/gio/xdgmime/xdgmime.c
@@ -467,6 +467,9 @@ xdg_mime_get_mime_type_for_data (const void *data,
 {
   const char *mime_type;
 
+  if (len == 0)
+    return XDG_MIME_TYPE_UNKNOWN;
+
   xdg_mime_init ();
 
   if (_caches)
-- 
1.7.9.5

